### PR TITLE
CDAP-2538: Handle extra whitespace in CLI input

### PR DESCRIPTION
**Pull request by jwang47**
Fix for https://issues.cask.co/browse/CDAP-2538

### CODE REVIEW

**Comment by awholegunch** [comment]
OK, but this is a problem in 3.0.0. Are you going to do a PR for that?

**Comment by jwang47** [comment]
This is not in the cdap repo. We'll have PRs to use the latest version of this common-cli after common-cli is released.

**Comment by awholegunch** [comment]
I read too quickly, and didn't notice this was in /common rather than /cdap.

File: common-cli/src/test/java/co/cask/common/cli/CommandSetTest.java [hunk 1]
```diff
@@ -108,6 +108,31 @@ private void assertThrows(Callable<Void> callable, Function<Exception, Void> exc
...
+      }
+
+      @Override
+      public String getDescription() {
+        return "Greets x";
+      }
+    };
+
+    CommandSet commandSet = new CommandSet<Command>(ImmutableList.of(greetCommand));
+    CommandMatch match = commandSet.findMatch("greet      user x");
```
Comment (anew) [inline common-cli/src/test/java/co/cask/common/cli/CommandSetTest.java:130]: Can you also test `" greet user x"`(space in front), `"greet user   x"` (multi space after user) and  `"greet user x "` (space after end of command)? (I can't find a way to make multi-space sequences appear in this comment...).

Comment (jwang47) [inline common-cli/src/test/java/co/cask/common/cli/CommandSetTest.java:130]: Added test cases and also added a fix for these cases.

**Comment by awholegunch** [comment]
Looks like the way to input multi-spaces in GitHub: put them in their own "code" block:

```
 "multi   spaces   "
```

Inline: `"multi   spaces          "` doesn't work.

**Comment by anew** [comment]
LGTM

**Pull request merged**